### PR TITLE
feat(screencast): add private recording APIs and basic test

### PR DESCRIPTION
**Pull request by yury-s**
#1158

### CODE REVIEW

File: src/page.ts [hunk 1]
```diff
@@ -58,6 +58,8 @@ export interface PageDelegate {
   canScreenshotOutsideViewport(): boolean;
   resetViewport(): Promise<void>; // Only called if canScreenshotOutsideViewport() returns false.
   setBackgroundColor(color?: { r: number; g: number; b: number; a: number; }): Promise<void>;
+  startVideoRecording(options: types.VideoRecordingOptions): Promise<void>;
```
Comment (pavelfeldman) [inline src/page.ts:61]: Let's move this to the browser context to cover popups, wdyt?

Comment (yury-s) [reply src/page.ts:61]: That's the ultimate plan. This requires generating file names, events that video started etc. For now I'd like to have basic functionality to provide some test coverage. Context level API can be added on top of it. Do you think we won't need page level start/stop functionality?

**Comment by aslushnikov** [comment]
I'd love to have a `page.screencast.*` namespace!

File: src/page.ts (hunk 1 above)
Comment (pavelfeldman) [reply src/page.ts:61]: As long as it is on delegate only - lgtm!

**Comment by pavelfeldman** [comment]
> I'd love to have a page.screencast.* namespace!

I actually had an argument against namespaces - they made rpc / other langs a bit harder to implement. So I would love to learn why you prefer it to make sure we make a balanced call

**Review by pavelfeldman — APPROVED** [review]

**Pull request merged**